modules/account-lib/src/coin/sol/transferBuilder.ts

```diff
@@ -1,37 +1,67 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { BuildTransactionError } from '../baseCoin/errors';
 import { TransactionBuilder } from './transactionBuilder';
 import { Transaction } from './transaction';
-import { Transaction as SolTransaction } from '@solana/web3.js';
+import { isValidAmount, isValidPublicKey } from './utils';
 import { TransactionType } from '../baseCoin';
+import { InstructionBuilderTypes } from './constants';
+import { Memo, Transfer } from './iface';
 
 export class TransferBuilder extends TransactionBuilder {
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
 
   protected get transactionType(): TransactionType {
-    return TransactionType.WalletInitialization;
+    return TransactionType.Send;
   }
 
-  /** @inheritdoc */
-  protected buildSolTransaction(): SolTransaction {
-    throw new NotImplementedError('buildImplementation not implemented');
+  /**
+   *  Set the memo
+   *
+   * @param {string} memo
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  memo(memo: string): this {
+    this.validateMemo(memo);
+    const memoData: Memo = {
+      type: InstructionBuilderTypes.Memo,
+      params: { memo },
+    };
+    this._instructionsData.push(memoData);
+    return this;
   }
 
-  /** @inheritdoc */
-  protected async buildImplementation(): Promise<Transaction> {
-    throw new NotImplementedError('buildImplementation not implemented');
-  }
+  /**
+   *  Set a transfer
+   *
+   * @param {string} fromAddress - the sender address
+   * @param {string} toAddress - the receiver address
+   * @param {string} amount - the amount sent
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  transfer(fromAddress: string, toAddress: string, amount: string): this {
```

Comment:
traditional builder pattern is to split this up into multiple functions to set each param. each func validates its own inputs.

e.g.
```
const solTransfer = new TransferBuilder()
    .from('fromAddress')
    .to('destAddress')
    .amount('100 SOL')
    .memo('hi there')
    .build();
```

Reply:
In order to be able to add multiple transfers i though it may be  better to set the all 3 transfer params (from, to, amount) in the same method.